facefusion/job_runner.py

```diff
@@ -0,0 +1,97 @@
+import os
+import shutil
+from typing import Dict
+from facefusion.ffmpeg import concat_video
+from facefusion.filesystem import is_video, create_temp, get_temp_file_path, is_directory, is_file
+from facefusion import logger, wording
+from facefusion.job_manager import read_job_file, set_step_status, move_job_file, get_job_ids, get_total_steps, get_step_status
+from facefusion.typing import JobStep, HandleStep
+
+
+def run_jobs(handle_step : HandleStep) -> None:
+	job_queued_ids = sorted(get_job_ids('queued'))
+
+	for job_id in job_queued_ids:
+		run_job(job_id, handle_step)
+		total_steps = get_total_steps(job_id)
+		completed_steps = 0
+
+		for step_index in range(total_steps):
+			if get_step_status(job_id, step_index) == 'completed':
+				completed_steps += 1
+		# TODO: logger break
+		logger.info(wording.get('job_processed').format(completed_steps = completed_steps, total_steps = total_steps, job_id = job_id), __name__.upper())
+
+
+def run_job(job_id : str, handle_step : HandleStep) -> bool:
+	job = read_job_file(job_id)
+	steps = job.get('steps')
+
+	if run_steps(job_id, steps, handle_step) and apply_merge_action(job_id):
+		return move_job_file(job_id, 'completed')
+	return move_job_file(job_id, 'failed')
+
+
+def run_steps(job_id : str, steps : list[JobStep], handle_step : HandleStep) -> bool:
+	for step_index, step in enumerate(steps):
+		step_args = step.get('args')
+		output_path = step_args.get('output_path')
+		target_path = step_args.get('target_path')
+
+		if not is_directory(output_path):
```

Comment:
this process flow is a bit hard to follow... it does a mutation rather than a run

if, if, if, else ... hard to read